### PR TITLE
Layers fixes

### PR DESCRIPTION
**Pull request by webbertakken**
# Layer fixes
- Redo: Layer initialisation
- Fixed: Map control not displaying whether or not overlays were enabled
- Added: Allow WMS layer as base layer
- Fixed: Previous base layers not being removed when adding another layer
- Fixed: Map control is now hidden if no layers are present
- Fixed: Attribution not being properly displayed for base layers

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
 
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation where applicable.

### CODE REVIEW

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:330`|
|---|
Pls add additional linefeed before comment

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:9`|
|---|
please add additional linefeed before comment

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:25`|
|---|
pls use '/**' here

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:29`|
|---|
Please add additional linefeed before comment

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:69`|
|---|
please use '/**' in this case

File: Resources/public/js/leaflet-embed.js [hunk 1]
```diff
@@ -80,6 +80,8 @@ CuriousMap.prototype.initialiseMapElements = function (options) {
   this.initiateMarker();
```
Comment (J063) [inline Resources/public/js/leaflet-embed.js]: |`line 147`|
|---|
Please add linefeed before comment

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:143`|
|---|
remove this linefeed or add comment

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:328`|
|---|
@{dcaa6089-4f82-45a6-9d94-cfc65106635b,Ben}

Still wondering why this is not simply layerDefinition.name? because the name group and control containers tend to cause confusion.

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:330`|
|---|
Add linefeed

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:483`|
|---|
@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}

Would it make sense to call this function only after all layers (fallback, base and overlay) have been added instead of calling it several times?

**Comment by webbertakken** [comment]
Refactored according to feedback, please review. @J063

**Pull request merged**